insights/combiners/os_release.py

```diff
@@ -0,0 +1,135 @@
+class OSRelease(object):
+    """
+    This combiner can be used to identify if the host an RHEL host by
+    checking the following parsers:
+
+        - :py:class:`insights.parsers.os_release.OsRelease`
+        - :py:class:`insights.parsers.redhat_release.RedhatRelease`
+        - :py:class:`insights.parsers.installed_rpms.InstalledRpms`
+
+    If none of the above parsers is available, NON-RHEL will be returned.
+
+    The :py:class:`insights.parsers.uname.Uname` parser is used to get the
+    booting kernel package.
+    """
+    def __init__(self, rpms, rhr, osr, uname):
+        self._is_rhel = False
+        self._product = "Unknown"
+        if osr:
+            names = list(filter(None,
+                                [osr.get('ID'), osr.get('NAME'),
+                                 osr.get('PRETTY_NAME')]))
+            if names and any(nr in nm.lower() for nm in names
+                                              for nr in NON_RHEL_KEYS):
+                self._product = names[-1]
+                # NON-RHEL Assertion 1: NG /etc/os-release
+                return
+        if rhr and any(nr in rhr.raw.lower() for nr in NON_RHEL_KEYS):
+            self._product = rhr.product
+            # NON-RHEL Assertion 2: NG /etc/redhat-release
+            return
+        if rpms and rpms.packages:
+            points = 0
+            installed_packages = 1  # kernel is must-installed
+            # check the booting 'kernel' first
+            boot_kn = InstalledRpm.from_package(
+                    'kernel-{0}'.format(uname.kernel)) if uname else None
+            if boot_kn:
+                # check the booting kernel only when Uname is available
+                for pkg in rpms.packages.get('kernel', []):
+                    if pkg == boot_kn:
+                        if (pkg.redhat_signed and
+                                pkg.vendor and pkg.vendor == 'Red Hat, Inc.'):
+                            points += 1
+                        break
+            # check other packages
+            for pkg_name in MIN_RHEL_PKGS:
+                pkg = rpms.newest(pkg_name)
+                if pkg:
+                    installed_packages += 1
+                    if (pkg.redhat_signed and
+                            pkg.vendor and pkg.vendor == 'Red Hat, Inc.'):
+                        points += 1
+            if points >= round(THRESHOLD * installed_packages):
+                # RHEL: more than THRESHOLD packages are from Red Hat
+                self._is_rhel = True
+                self._product = RHEL_STR
+        elif ((rhr and rhr.is_rhel) or
+                (osr and names and
+                 any(k in v.lower() for k in RHEL_KEYS for v in names))):
+            # backup plan when "rpm" command is corrupt or blocked
+            # RHEL: when `os-release` *OR* `redhat-release` are for RHEL
+            self._is_rhel = True
+            self._product = RHEL_STR
+        # else: Unknown OS Product
+
+    @property
```

Comment:
Why have properties that just return an attribute of the object?  Why not just set the `self.is_rhel` and `self.product` properties in the `__init__`?